src/components/SigninPopup/SigninPopup.js

```diff
@@ -1,25 +1,128 @@
 import React, { useState, useEffect } from "react";
 import PopupWithForm from '../PopupWithForm/PopupWithForm';
+import * as auth from '../../utils/auth';
 
 function SigninPopup(props) {
 
     const [email, setEmail] = useState('');
     const [password, setPassword] = useState('');
+    const [emailErrorMessage, setEmailErrorMessage] = useState('');
+    const [passwordErrorMessage, setPasswordErrorMessage] = useState('');
+    const [isEmailValid, setIsEmailValid] = useState(false);
+    const [isPasswordValid, setIsPasswordValid] = useState(false);
+    const [isFormValid, setIsFormValid] = useState(false)
+
+    const resetSigninForm = () => {
+        setEmail('');
+        setPassword('');
+        setEmailErrorMessage('');
+        setPasswordErrorMessage('');
+        setIsFormValid(false);
+        setIsPasswordValid(false);
+        setIsEmailValid(false);
+        console.log(isEmailValid, isPasswordValid)
+    };
+
+    useEffect(() => {
+        resetSigninForm();
+    }, [props.isOpen]);
+
+    function validateEmail(value) {
+        if (!value || !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)) {
+            setEmailErrorMessage('Invalid email address');
+            setIsEmailValid(false);
+        } else {
+            setIsEmailValid(true);
+            setEmailErrorMessage('');
+        }
+    }
+
+    function validatePassword(value) {
+        if (!value || (value.length + 1) < 6) {
+            console.log(value.length)
+            setPasswordErrorMessage('Password should have at least 6 characters');
+            setIsPasswordValid(false);
+        } else {
+            setIsPasswordValid(true);
+            setPasswordErrorMessage('');
+        }
+    };
+
+    function validateForm() {
+        setIsFormValid(false);
+        // console.log(isEmailValid, isPasswordValid)
+        if (isEmailValid && isPasswordValid) {
+            setIsFormValid(true);
+        } else {
+            setIsFormValid(false);
+            // console.log('some input is invalid')
+        } 
+    }
 
     function handleSubmit(e) {
-        e.preventDefault();
-        props.onSubmit()
-       
+        if (!isFormValid) {
+            e.preventDefault();
+            props.handleError();
+        } else {
+            e.preventDefault();
+            auth.authorize(email, password)
```

Comment:
**Might be better**

it's better to make api requests only in `App` because `Login` and `Register` should not know how to do it. They should be independent from such logics